wordpress-mobile/WordPress-Android
# My Site Dashboard-Phase 2: Refactor Builder parameter classes

Parent #15215

This PR refactors `CardBuilder` so that each card type uses its own parameters object which is passed via `MySiteViewModel`. This PR only includes refactoring, there has been no new functionality added.

Included in this PR are the following cards types:
- Domain Registration
- Quick Actions
- Quick Start
- Post
- Site Info

Not included
`DynamicCardsBuilder` and `SiteItemsBuilder`. They will be addressed separately.

Existing tests have been refactored to account for the new params classes.

**Notes**
- The best way to review this PR is to go commit-by-commit. Deteket/ktlint and further refactoring come later on in the commit history. 
- I had one case in which I was undecided on a refactor, that was in `CardsBuilder` and possibly removing all the single line functions and placing the logic directly in the build. I was hesitant because I'm not sure if we are going to be adding any tracking before building. I'm open to it and can make the change before this PR is merged. Let me know.
- These changes are not controlled by a feature flag, and will merge to develop as is. Given this is only the first week of the sprint, we should be able to test thoroughly before the next version moves to beta. ⚠️ 

**To test:**
- Smoke test the app and make sure the cards are displayed as expected.
- The post card is still behind a feature flag, so test with the flag on and off
- Quick Actions - should be visible for WordPress and hidden for Jetpack
- Site Info Card - all interactions work as expected
- Quick Start - all interactions work as expected (and when the quickStartDynamicCardsFeatureConfig is on, then this card doesn't show)
- Domain Registration -  (Visible for paid plan site when SiteDomainFeatureConfig flag enabled, hidden otherwise.)


## Regression Notes
1. Potential unintended areas of impact
These changes are not controlled by a feature flag, and will merge to develop as is. Given this is only the first week of the sprint, we should be able to test thoroughly before the next version moves to beta. ⚠️

2. What I did to test those areas of impact (or what existing automated tests I relied on) N/A

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:
